test/GraphStoreProtocolTest.cpp

```diff
@@ -0,0 +1,181 @@
+      GraphStoreProtocol::transformPost(
+          ad_utility::testing::MakePostRequest(
+              "/?default", "application/sparql-results+xml", ""),
+          DEFAULT{}),
+      testing::HasSubstr(
+          "Mediatype \"application/sparql-results+xml\" is not supported for "
+          "SPARQL Graph Store HTTP Protocol in QLever."));
+  AD_EXPECT_THROW_WITH_MESSAGE(
+      GraphStoreProtocol::transformPost(
+          ad_utility::testing::MakePostRequest(
+              "/?default", "application/n-quads", "<a> <b> <c> <d> ."),
+          DEFAULT{}),
+      testing::HasSubstr("Not a single media type known to this parser was "
+                         "detected in \"application/n-quads\"."));
+  AD_EXPECT_THROW_WITH_MESSAGE(
+      GraphStoreProtocol::transformPost(
+          ad_utility::testing::MakePostRequest(
+              "/?default", "application/unknown", "fantasy"),
+          DEFAULT{}),
+      testing::HasSubstr("Not a single media type known to this parser was "
+                         "detected in \"application/unknown\"."));
+}
+
+TEST(GraphStoreProtocolTest, transformGet) {
+  auto expectTransformGet =
+      [](const ad_utility::httpUtils::HttpRequest auto& request,
+         const testing::Matcher<const ParsedQuery&>& matcher,
+         ad_utility::source_location l =
+             ad_utility::source_location::current()) {
+        auto trace = generateLocationTrace(l);
+        const ad_utility::url_parser::ParsedUrl parsedUrl =
+            ad_utility::url_parser::parseRequestTarget(request.target());
+        const GraphOrDefault graph =
+            GraphStoreProtocol::extractTargetGraph(parsedUrl.parameters_);
+        EXPECT_THAT(GraphStoreProtocol::transformGet(graph), matcher);
+      };
+  expectTransformGet(
+      ad_utility::testing::MakeGetRequest("/?default"),
+      m::ConstructQuery({{Var{"?s"}, Var{"?p"}, Var{"?o"}}},
+                        m::GraphPattern(matchers::Triples({SparqlTriple(
+                            TC(Var{"?s"}), "?p", TC(Var{"?o"}))}))));
+  expectTransformGet(
+      ad_utility::testing::MakeGetRequest("/?graph=foo"),
+      m::ConstructQuery({{Var{"?s"}, Var{"?p"}, Var{"?o"}}},
+                        m::GraphPattern(m::GroupGraphPatternWithGraph(
+                            {}, t::iri("<foo>"),
+                            matchers::Triples({SparqlTriple(
+                                TC(Var{"?s"}), "?p", TC(Var{"?o"}))})))));
+}
+
+TEST(GraphStoreProtocolTest, transformGraphStoreProtocol) {
+  EXPECT_THAT(GraphStoreProtocol::transformGraphStoreProtocol(
+                  ad_utility::testing::MakeGetRequest("/?default")),
+              m::ConstructQuery({{Var{"?s"}, Var{"?p"}, Var{"?o"}}},
+                                m::GraphPattern(matchers::Triples({SparqlTriple(
+                                    TC(Var{"?s"}), "?p", TC(Var{"?o"}))}))));
+  EXPECT_THAT(
+      GraphStoreProtocol::transformGraphStoreProtocol(
+          ad_utility::testing::MakePostRequest(
```

Comment:
A `using namespace ad_utiility::testing` inside this function would make the code a little shorter (not too important though).